src/main/java/rx/internal/operators/OperatorSwitchIfEmpty.java

```diff
@@ -38,33 +38,35 @@ public OperatorSwitchIfEmpty(Observable<? extends T> alternate) {
         final SwitchIfEmptySubscriber parent = new SwitchIfEmptySubscriber(child, ssub);
         ssub.set(parent);
         child.add(ssub);
+        child.setProducer(new Producer() {
```

Comment:
I'd instantiate the ProducerArbiter before ```parent``` and supply it to ```SwitchIfEmptySubscriber``` and ```child```, saving on this 'relaying' allocation. (Remark, our life would be much easier if we could implement ```Producer``` in ```Subscriber```, but since ```request()``` is ```protected final```, it is unoverridable :( )